subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/versions/AndroidGradlePluginVersions.groovy

```diff
@@ -131,21 +131,26 @@ class AndroidGradlePluginVersions {
     }
 
     private static JavaVersion getMinimumJavaVersionFor(VersionNumber agpVersion) {
-        if (agpVersion.baseVersion < AGP_7_0_VERSION_NUMBER) {
+        if (agpVersion.baseVersion < AGP_7_0) {
             return JavaVersion.VERSION_1_8
         }
         return JavaVersion.VERSION_11
     }
 
     private static JavaVersion getMaximumJavaVersionFor(VersionNumber agpVersion) {
         // This is mainly to prevent running all AGP tests on too many java versions and reduce CI time
-        if (agpVersion.baseVersion < AGP_7_0_VERSION_NUMBER) {
+        if (agpVersion.baseVersion < AGP_7_0) {
             return JavaVersion.VERSION_11
         }
         return null
     }
 
-    static void assumeAgpSupportsKotlinVersion(String agpVersion, String kotlinVersion) {
+    static void assumeAgpSupportsCurrentJavaVersionAndKotlinVersion(String agpVersion, String kotlinVersion) {
+        assumeCurrentJavaVersionIsSupportedBy(agpVersion)
```

Comment:
Shouldn't this method be made `private` too?

Reply:
No, this method is used in different places without the other one when you have a pure Android project which doesn't use Kotlin.